### PR TITLE
Register the StorkClientRequestFilter when we detect a URI starting with stork:// or storks://

### LINKED ISSUE #28292: `StorkClientRequestFilter` not registered when using `ClientBuilder` to build a jax-rs client
### Describe the bug

When using Stork with a manually-created jax-rs client (via `ClientBuilder.newClient()`), the `StorkClientRequestFilter` is not automatically registered.

Stork worked fine with the jax-rs client in Quarkus 2.12, but in 2.13 this is broken unless the user manually registers the filter.

See https://github.com/quarkusio/quarkus-super-heroes/blob/main/rest-fights/src/main/java/io/quarkus/sample/superheroes/fight/client/VillainClient.java#L38

If the user does not manually register the filter then they will end up with a failure that looks something like

```
java.lang.IllegalArgumentException: Invalid REST Client URL used: 'stork://villain-service/api/villains/random'
```

At a very minimum, this should be documented in the Quarkus 2.13 release notes.

### Expected behavior

I would expect Stork to "just work" with the jax-rs client, as it did prior to Quarkus 2.12.

### Actual behavior

The application fails if the user does not manually register the filter.

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

The `OpenTelemetryClientFilter` seems to be registered automatically when creating the jax-rs client this way, so there must be a way to do this as well for the `StorkClientRequestFilter`.

### PR DESCRIPTION
**Pull request by cescoffier**
Fix https://github.com/quarkusio/quarkus/issues/28292

### CODE REVIEW

**Review by geoand — APPROVED** [review]
We should probably move this logic to build time as a next step, but it's fine for now to overcome the original issue

**Review by Sgitario — COMMENTED** [review]
The changes look good to me, however I think it can easily be covered by reusing this test: https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/stork/StorkDevModeTest.java

**Review by aureamunoz — APPROVED** [review]
I don't know very well this part about filters so this LGTM.
Shouldn't we add a test also?

Looking to the CI fails

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/WebTargetImpl.java [hunk 1]
```diff
@@ -320,10 +320,44 @@ private void abortIfClosed() {
...
     }
 
+    /**
+     * If the URI starts with stork:// or storks://, then register the StorkClientRequestFilter automatically.
+     *
+     * @param configuration the configuration
+     * @param actualUri the uri
+     */
+    private static void registerStorkFilterIfNeeded(ConfigurationImpl configuration, URI actualUri) {
+        if (actualUri.getScheme().startsWith("stork") && !isStorkAlreadyRegistered(configuration)) {
```
Comment (aureamunoz) [inline independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/WebTargetImpl.java:336]:
```suggestion
        if (actualUri.getScheme() != null && actualUri.getScheme().startsWith("stork") && !isStorkAlreadyRegistered(configuration)) {
```

Comment (aureamunoz) [reply independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/WebTargetImpl.java:336]: This fixes the InvalidURITest failing currently

**Review by aureamunoz — CHANGES_REQUESTED** [review]

**Comment by cescoffier** [comment]
Sorry got caught up by another urgent matter. I will have a look at the issues tomorrow morning.

(And here goes my blog post...)

**Comment by cescoffier** [comment]
@Sgitario I added some tests.

**Review by Sgitario — APPROVED** [review]
lgtm!

**Pull request merged**